python_da/dq_checks/src/app.py

```diff
@@ -58,17 +60,50 @@ def download_blob_csv_data():
         Exception: For general exceptions that might occur while trying to download the blob data.
     """
 
+def download_blob_csv_data():
+    """
+    Downloads the CSV from Azure Blob Storage, converts it into a Pandas DataFrame, and returns it.
+
+    This function initializes a BlobServiceClient using the connection credentials, retrieves the blob client for a specific
+    container, and lists all blobs within that container. 
+    
+    FOR TESTING PURPOSES!!!!!If the downloaded CSV data is empty or contains only whitespace,
+    it will attempt to download 'dirty_startup_dataset.csv' instead. # CHANGE ASAP AFTER FRIDAY DEMO
+
+    Logging and error handling is utilized to provide information about the process
+    and to provide specific errors. 
+
+    Returns:
+        pd.DataFrame: A Pandas DataFrame containing the CSV data downloaded from Azure Blob Storage.
+
+    Raises:
+        AzureError: If there is an Azure-specific error while trying to download the blob data.
+        Exception: For general exceptions that might occur while trying to download the blob data.
+    """
+
     try:
         logger.info('Initializing BlobServiceClient')
         blob_service_client = BlobServiceClient.from_connection_string(connection_string)
         container_client = blob_service_client.get_container_client(container_name)
         blob_list = container_client.list_blobs()
         blob_name = next(blob_list).name  # Assumes only one blob
 
+        if blob_name is None:  # Checking if any blobs in container.
+            logger.error('No blobs found in container')
+            raise Exception('No blobs found in the container')
+
         logger.info(f'Downloading CSV data from Azure Blob Storage: {blob_name}')
         blob_client = blob_service_client.get_blob_client(container_name, blob_name)
         blob_data = blob_client.download_blob()
         csv_data = blob_data.readall().decode('utf-8')
+
+        if not csv_data.strip(): 
+            logger.warning("Downloaded CSV data is empty or contains only whitespace. Attempting to download 'dirty_startup_dataset.csv' instead.")
+            blob_name = "dirty_startup_dataset.csv"
```

Comment:
can you declare the variable on top

Reply:
Yeah will do, I'll be doing a refactoring next week where I will remove all this stuff. It was mainly for testing on my machine.

Reply:
Resolved with latest commit.